core/executor/src/precompiles/modexp.rs

```diff
@@ -82,33 +107,24 @@ struct LargeNumber {
 }
 
 impl LargeNumber {
-    const MAX_NUM_SIZE: usize = 1024;
-
     fn parse(input: &[u8]) -> Result<Self, PrecompileFailure> {
-        let input_len = input.len();
-        if input_len < 96 {
-            return err!("Input length must be at least 96");
-        };
-
-        let base_size = Self::parse_big_uint(&input[0..32], true)?.unwrapped_as::<usize>();
-        let exponent_size = Self::parse_big_uint(&input[32..64], true)?.unwrapped_as::<usize>();
-        let modulo_size = Self::parse_big_uint(&input[64..96], true)?.unwrapped_as::<usize>();
+        let base_size = get_data(input, 0, 32)?.unwrapped_as::<usize>();
```

Comment:
Doesn't this panic when the `Integer` cannot fit in a `usize`? Is it OK to panic in precompiled contracts?